File: trial/src/main/java/gq/emiliodallatorre/trial/CustomAdaptor.java
package gq.emiliodallatorre.trial;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.StrictMode;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;


/**
 * @author Emilio Dalla Torre.
 */

public class CustomAdaptor extends BaseAdapter {

    private Context context;
    private List<Model> modelList;
    StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();


    public CustomAdaptor(Context context, List<Model> modelList) {
        this.context = context;
        this.modelList = modelList;

    }

    @Override
    public int getCount() {
        return modelList.size();
    }

    @Override
    public Object getItem(int i) {
        return modelList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        StrictMode.setThreadPolicy(policy);

        LinearLayout rootView = (LinearLayout) ((Activity) context).getLayoutInflater().inflate(R.layout.custom_list, null);
        final TextView date_and_creator = (TextView) rootView.findViewById(R.id.date_and_creator);
        final TextView title = (TextView) rootView.findViewById(R.id.title);
//        final ImageView image = (ImageView) rootView.findViewById(R.id.resim);
//        final TextView description = (TextView) rootView.findViewById(R.id.description);
        final Model model = modelList.get(i);

        Date date = new Date(model.getDate());

        DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
        title.setText(model.getTitle());
        date_and_creator.
                setText(String.format("%s | %s   |   %s", String.format("%02d:%02d", date.getHours(), date.getMinutes()), dateFormat.format(date), model.getCreator()));
/*        Bitmap bitmap = model.getImage();
        image.setImageBitmap(bitmap);
        description.setText(description.toString());
*/
        rootView.setOnClickListener(new View.OnClickListener() {
            public void onClick(View view) {
                String link = (model.getLink());
                String title = (model.getTitle());
                String creator = (model.getCreator());
                Intent intent = new Intent(context, webb.class);
                intent.putExtra("link", link);
                intent.putExtra("title", title);
                intent.putExtra("creator", creator);
                context.startActivity(intent);
            }

        });

        rootView.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent motionEvent) {
                return false;
            }
        });

        return rootView;
    }
}

File: app/src/main/java/it/LaVocedelBrunoFranchetti/rssreader/CustomAdaptor.kt
package it.LaVocedelBrunoFranchetti.rssreader

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.StrictMode
import android.support.constraint.ConstraintLayout
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import java.text.SimpleDateFormat
import java.util.*



class CustomAdaptor internal constructor(private val context: Context, private val modelList: ArrayList<Model>) : BaseAdapter() {

    private var policy: StrictMode.ThreadPolicy = StrictMode.ThreadPolicy.Builder().permitAll().build()
    private val rootView = (context as Activity).layoutInflater.inflate(R.layout.custom_list, null) as ConstraintLayout
    private val TAG = "CustomAdaptor"

    private val date_and_creatorTW = rootView.findViewById(R.id.date_and_creator) as TextView
    private val titleTW = rootView.findViewById(R.id.title) as TextView

    override fun getCount(): Int {
        return modelList.size
    }

    override fun getItem(i: Int): Any {
        return modelList[i]
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        StrictMode.setThreadPolicy(policy)

        val model = modelList[i]
        val title = model.title
        val creator = model.creator
        val description = model.description
        val date = Date(model.date)
        val dateandcreator: String = (String.format("%02d:%02d", date.hours, date.minutes) + "   |   " + SimpleDateFormat("%02d:%02d | dd.MM.yyyy", Locale.getDefault()).format(date) + "   |   " + model.creator)

        date_and_creatorTW.text = dateandcreator
        titleTW.text = title

        rootView.setOnClickListener {
            val link = model.link
            val intent = Intent(context, ArticleView::class.java)
            intent.putExtra("link", link)
            intent.putExtra("title", title)
            intent.putExtra("creator", creator)
            context.startActivity(intent)
        }

        // rootView.setOnTouchListener { view, motionEvent -> false }

        Log.d(TAG, dateandcreator)
        return rootView
    }
}